### DISCUSSION TITLE
Get started with GraphQL in .NET Core - Does not work when followed directly as written - CORS

### DISCUSSION THREAD

**Discussion by trimeloni** [ChilliCream/hotchocolate#7288] [General]
The `Get started with GraphQL in .NET Core`  page ( https://chillicream.com/docs/hotchocolate/v13/get-started-with-graphql-in-net-core )  directions do not work when following along with them.

It looks like the main issue is that `Banana Cake Pop` used to run right from `localhost` - note I can see this from the `Getting Started` video at the top - but now is a redirect to `https://cdn.bananacakepop.com/web/latest/`.   Unfortunately, out of the box with these directions, you can not load the schema or anything else because you get a CORS error.

I think in many environments you won't need to worry about CORS and it's to bad to have to add it to the super short example given.  It seems like either the commands to allow CORS - from anywhere or the specific `bananacakepop.com` site - should be added but noted in a way that calls out that this isn't something you need to add in all environments.  Or, have Banana Cake Pop run from `localhost` again.  I'm assuming though it was a conscience decision to have it always go to the website to get the latest.

I'd would just hate to see people give up early because they think that the simple getting started doesn't work.

**Comment by wavedeck** [comment]
this does not seem right.. I'm on v14.0.0-p135 and Banana Cake Pop runs entirely locally, what version do you use?

**Reply by wavedeck** [reply]
this is also the case for the latest stable v13.9.8

**Reply by trimeloni** [reply]
Hello, sorry for the delay, I was out of town for a bit.

I went ahead and did this again from scratch and tried to make sure I didn't do anything extra.  I took some screen shots as well as added a text version for easier review.  I pretty much left everything in here since it was short, unless it had some identifying data.

I also edited the Program.cs in VS Code to try to make sure nothing extra could have been changing files around.

DotNet: 8.0.303
HotChocolate.AspNetCore: 13.9.9  (what I got from just following the page exactly without specifying a version)

```
>dotnet --version
8.0.303

------------------------------------------------------------------

>type Program.cs

var builder = WebApplication.CreateBuilder(args);

builder.Services
    .AddGraphQLServer()
    .AddQueryType<Query>();

var app = builder.Build();

app.MapGraphQL();

app.Run();



public class Book
{
    public string Title { get; set; }

    public Author Author { get; set; }
}

public class Author
{
    public string Name { get; set; }
}

public class Query
{
    public Book GetBook() =>
        new Book
        {
            Title = "C# in depth.",
            Author = new Author
            {
                Name = "Jon Skeet"
            }
        };
}

------------------------------------------------------------------
>type Demo.csproj

<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="HotChocolate.AspNetCore" Version="13.9.9" />
  </ItemGroup>

</Project>

------------------------------------------------------------------
>dotnet run

Building...
Program.cs(26,19): warning CS8618: Non-nullable property 'Name' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [Demo.csproj]
Program.cs(19,19): warning CS8618: Non-nullable property 'Title' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [Demo.csproj]
Program.cs(21,19): warning CS8618: Non-nullable property 'Author' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [Demo.csproj]
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://localhost:5157
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\Temp\HotChocTemp\Demo
info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down...
```

Image of the Redirect happening within FireFox Network Tools ( ran in incognito mode).
![image](https://github.com/user-attachments/assets/502df1b8-2b2c-47b6-9edb-b12f7cf912b1)

Running 
![image](https://github.com/user-attachments/assets/fcfe2016-4ce4-4ea1-a685-adc787dad956)
